app/team.go

```diff
@@ -480,43 +480,10 @@ func (a *App) sendUpdatedMemberRoleEvent(userID string, member *model.TeamMember
 }
 
 func (a *App) AddUserToTeam(teamID string, userID string, userRequestorId string) (*model.Team, *model.AppError) {
-	tchan := make(chan store.StoreResult, 1)
-	go func() {
-		team, err := a.Srv().Store.Team().Get(teamID)
-		tchan <- store.StoreResult{Data: team, NErr: err}
-		close(tchan)
-	}()
-
-	uchan := make(chan store.StoreResult, 1)
-	go func() {
-		user, err := a.Srv().Store.User().Get(userID)
-		uchan <- store.StoreResult{Data: user, NErr: err}
-		close(uchan)
-	}()
-
-	result := <-tchan
-	if result.NErr != nil {
-		var nfErr *store.ErrNotFound
-		switch {
-		case errors.As(result.NErr, &nfErr):
-			return nil, model.NewAppError("AddUserToTeam", "app.team.get.find.app_error", nil, nfErr.Error(), http.StatusNotFound)
-		default:
-			return nil, model.NewAppError("AddUserToTeam", "app.team.get.finding.app_error", nil, result.NErr.Error(), http.StatusInternalServerError)
-		}
-	}
-	team := result.Data.(*model.Team)
-
-	result = <-uchan
-	if result.NErr != nil {
-		var nfErr *store.ErrNotFound
-		switch {
-		case errors.As(result.NErr, &nfErr):
-			return nil, model.NewAppError("AddUserToTeam", MissingAccountError, nil, nfErr.Error(), http.StatusNotFound)
-		default:
-			return nil, model.NewAppError("AddUserToTeam", "app.user.get.app_error", nil, result.NErr.Error(), http.StatusInternalServerError)
-		}
+	user, team, err := a.prepareInviteNewUsersToTeam(teamID, userID)
```

Comment:
So as I see it, all of these changes basically make these calls synchronous. These actions aren't in the hot path, so it should be fine. But I am not sure if I would call it "unnecessary". They are just written in this way because we are checking for the error values one after the other.

But if we did something like:

```go
g := new(errgroup.Group)
var a,b Item
  g.Go(func() error {
  	var err error
  	a, err = Foo()
  	return err
	})

g.Go(func() error {
  	var err error
  	b, err = Bar()
  	return err
	})

g.Wait()
```

Then the intention is much more clear.

Reply:
Yeah we can defo do that, the usage of the word "unnecessary" maybe is overpowering here or incorrect, but my sole intention with these was the fact that in case of an error on the top level query we are unnecessarily fetching everything else, also the code does not read really well, with all the channel wait calls.

Reply:
> in case of an error on the top level query we are unnecessarily fetching everything else, 

Yeah, so that will be the case for all concurrent calls that we are doing. I think maybe you wanted to say that our "implementation" of concurrent calls is not very readable with the way we pass errors with channels and all?

That is very true and I completely agree with it. We can move to using the `errgroup` pattern, which makes it much more readable IMO. But making all of them synchronous is a different solution I think.

Reply:
Also, your first PR fixed a different problem. In that case, we were completely discarding the second query when we got the results from the first one. That case is a completely valid case of conditionally issuing synchronous calls because that's the happy path. But these are just normal concurrent calls from all of which, we need the results.

Reply:
Totally, my only view here, is if those calls need to be async in the first place. Like if all take tens of ms, a user will not notice the difference, between promise-based calls and sync calls. But you mentioned that some of those are heavy once, so let me profile them and see what we can do there. Maybe just optimising the queries will be a good start.

Reply:
Ah yes, optimising those queries definitely sounds like a good plan. Thanks John!